src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -423,6 +600,15 @@ class StubGenerator: public StubCodeGenerator {
 
   address generate_catch_exception() {
     StubId stub_id = StubId::stubgen_catch_exception_id;
+    int entry_count = StubInfo::entry_count(stub_id);
+    assert(entry_count == 1, "sanity check");
+    if (find_archive_data(stub_id)) {
+      address start = nullptr;
+      address end = nullptr;
+      load_archive_data(stub_id, start, end);
+      return start;
+    }
+
```

Comment:
This code pattern repeats a lot. Can we move it into `load_archive_data()`?
```
address start = load_archive_data(stub_id);
if (start != nullptr) {
  return start;
}
```

And have an other specialized `load_archive_data` if you need `end` value.